lib/OpenQA/WebAPI/GruJob.pm

```diff
@@ -49,8 +50,23 @@ sub execute {
           && exists $self->info->{notes}{gru_id};
     }
     else {
+        # here we call finish, which will either actually finish the job or
+        # retry it
         $self->finish(defined $buffer ? $buffer : 'Job successfully executed');
-        $self->_delete_gru($self->info->{notes}{gru_id}) if exists $self->info->{notes}{gru_id};
+        my $state = $self->info->{state};
+        my $jobid = $self->id;
+        my $gruid;
+        $gruid = $self->info->{notes}{gru_id} if exists $self->info->{notes}{gru_id};
```

Comment:
You're triple requesting `$self->info` here, that's two extra Postgres roundtrips for the whole info data structure.

Reply:
oh, fair point. will fix.